coredns/endpointslice/map.go

```diff
@@ -86,13 +93,14 @@ func (m *Map) GetDNSRecords(hostname, cluster, namespace, name string, checkClus
 	}
 }
 
-func NewMap() *Map {
+func NewMap(localClusterID string) *Map {
 	return &Map{
-		epMap: make(map[string]*endpointInfo),
+		epMap:          make(map[string]*endpointInfo),
+		localClusterID: localClusterID,
 	}
 }
 
-func (m *Map) Put(es *discovery.EndpointSlice) {
+func (m *Map) Put(es *discovery.EndpointSlice, localClient v1.DiscoveryV1Interface) {
```

Comment:
I think it's cleaner to pass `localClient` into the `NewMap` ctor and store in `Map`.